packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/camera.dart

```diff
@@ -155,49 +180,154 @@ class CameraUpdate {
   }
 
   /// Returns a camera update that modifies the camera zoom level by the
   /// specified amount. The optional [focus] is a screen point whose underlying
   /// geographical location should be invariant, if possible, by the movement.
   static CameraUpdate zoomBy(double amount, [Offset? focus]) {
-    if (focus == null) {
-      return CameraUpdate._(<Object>['zoomBy', amount]);
-    } else {
-      return CameraUpdate._(<Object>[
-        'zoomBy',
-        amount,
-        <double>[focus.dx, focus.dy],
-      ]);
-    }
+    return CameraUpdateZoomBy(amount, focus);
   }
 
   /// Returns a camera update that zooms the camera in, bringing the camera
   /// closer to the surface of the Earth.
   ///
   /// Equivalent to the result of calling `zoomBy(1.0)`.
   static CameraUpdate zoomIn() {
-    return const CameraUpdate._(<Object>['zoomIn']);
+    return const CameraUpdateZoomIn();
   }
 
   /// Returns a camera update that zooms the camera out, bringing the camera
   /// further away from the surface of the Earth.
   ///
   /// Equivalent to the result of calling `zoomBy(-1.0)`.
   static CameraUpdate zoomOut() {
-    return const CameraUpdate._(<Object>['zoomOut']);
+    return const CameraUpdateZoomOut();
   }
 
   /// Returns a camera update that sets the camera zoom level.
   static CameraUpdate zoomTo(double zoom) {
-    return CameraUpdate._(<Object>['zoomTo', zoom]);
+    return CameraUpdateZoomTo(zoom);
   }
 
-  final Object _json;
-
   /// Converts this object to something serializable in JSON.
-  Object toJson() => _json;
+  Object toJson();
+}
+
+/// Defines a camera move to a new position.
+class CameraUpdateNewCameraPosition extends CameraUpdate {
+  /// Creates a camera move.
+  const CameraUpdateNewCameraPosition(this.cameraPosition)
+      : super._(CameraUpdateType.newCameraPosition);
+
+  /// The new camera position.
+  final CameraPosition cameraPosition;
+  @override
+  Object toJson() => <Object>['newCameraPosition', cameraPosition.toMap()];
+}
+
+/// Defines a camera move to a latitude and longitude.
+class CameraUpdateNewLatLng extends CameraUpdate {
+  /// Creates a camera move to latitude and longitude.
+  const CameraUpdateNewLatLng(this.latLng)
+      : super._(CameraUpdateType.newLatLng);
+
+  /// New latitude and longitude of the camera..
+  final LatLng latLng;
+  @override
+  Object toJson() => <Object>['newLatLng', latLng.toJson()];
+}
+
+/// Defines a camera move to a new bounding latitude and longitude range.
+class CameraUpdateNewLatLngBounds extends CameraUpdate {
+  /// Creates a camera move to a bounding range.
+  const CameraUpdateNewLatLngBounds(this.bounds, this.padding)
+      : super._(CameraUpdateType.newLatLngBounds);
+
+  /// The northeast and southwest bounding coordinates.
+  final LatLngBounds bounds;
+
+  /// The amount of padding by which the view is inset.
+  final double padding;
+  @override
+  Object toJson() => <Object>['newLatLngZoom', bounds.toJson(), padding];
```

Comment:
Thanks for flagging that; I've retracted 2.9.1 on pub.dev. I assumed we already had Dart unit testing of the JSON we sent, since we usually do, but apparently not in this case.

@yaakovschectman You'll need to fix this so we can publish a new non-retracted version of the platform interface to unblock your other PR.

Reply:
I've added you both as reviewers to a PR that addresses this fix.

Reply:
Why can't I find the new version (2.9.1) of the google_maps_flutter package in pub.dev ?

Reply:
As far as I know, `google_maps_flutter` shouldn't have a new version yet. `google_maps_flutter_platform_interface` is now at version 2.9.2